utils/log-collector

```diff
@@ -0,0 +1,101 @@
+    case ${opt} in
+        s ) 
+            stackName=$OPTARG
+            ;;
+        i )
+            instanceId=$OPTARG
+            ;;
+        o )
+            outputFormat=$OPTARG
+            ;;
+        h )
+            usage
+            exit 0
+            ;;
+        \? )
+            echo "Invalid Option: -$OPTARG" 1>&2
+            exit 1
+            ;;
+        : )
+            echo "Invalid option: $OPTARG requires an arguemnt" 1>&2
+            ;;
+    esac
+done
+shift $((OPTIND -1))
+
+now=$(date +"%F")
+outputFormat="${outputFormat:=text}"
+
+# Create a temporary directory to store our logs
+pwd=$(pwd)
+dir=$(mktemp -d "$pwd"/buildkite-logs-"$now")
+
+echo "Collecting logs for instance $instanceId"
+
+# Collect the logs for the instance
+aws logs get-log-events --log-group-name /buildkite/buildkite-agent --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/buildkite-agent-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/system --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/system-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/lifecycled --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/lifecycled-"$instanceId".log
+aws logs get-log-events --log-group-name /buildkite/docker-daemon --log-stream-name "$instanceId" --output "$outputFormat" >> "$dir"/docker-daemon-"$instanceId".log
+
+# CloudWatch stores the creationTime as milliseconds since epoch UTC. 
+# Since macOS doesn't support calculating the time since epoch in milliseconds, we don't have perfect accuracy on this, but it should be as close to 24 hours as possible
+# Currently only collects the last 24h of logs, but it might be good to be able to specify a different value for this.
+logAge=$(date -v-1d "+%s000")
+
+# Collect some information about the Lambda function and associated log groups
+# Change the timeframe to be dynamically set with a default of 24h. Currently collects the last 24 hours of Lambda logs
+lambdaFunction=$(aws cloudformation describe-stack-resources --stack-name "$stackName" --logical-resource-id Autoscaling --query "StackResources[*].PhysicalResourceId" --output text | sed -nr 's/.*stack\/(.*)\/.*/\1/p')
+lambdaLogGroup=$(aws cloudformation describe-stack-resources --stack-name "$lambdaFunction" --logical-resource-id LogGroup --query "StackResources[*].PhysicalResourceId" --output text)
+logStreams=$(aws logs describe-log-streams --log-group-name "$lambdaLogGroup" --order-by LastEventTime --query "logStreams[?creationTime > \`$logAge\`] | [*].logStreamName" --output text | tr -s '[:space:]' '\n')
+
+# Iterate through all of the log streams we collected from the lambda log group
+
+echo "Collecting Lambda logs for $lambdaFunction"
+for i in $logStreams
+do
+    fileName=$(echo "$i" | sed -r 's/\//-/g')
+    aws logs get-log-events --log-group-name "$lambdaLogGroup" --log-stream-name "$i" --output "$outputFormat" >> "$dir"/lambda-"$fileName".log
```

Comment:
I personally find this style easier to read:

```diff
-"$dir"/lambda-"$fileName".log
+"$dir/lambda-$fileName.log"
```

Or if you're worried about differentiating the vars from other content:

```diff
-"$dir"/lambda-"$fileName".log
+"${dir}/lambda-${fileName}.log"
```